modules/geo-layers/src/mesh-layer/mesh-layer.js

```diff
@@ -47,19 +67,24 @@ export default class _MeshLayer extends SimpleMeshLayer {
   }
 
   getModel(mesh) {
-    const pbrMaterial = this.props.pbrMaterial;
+    const {id, pickFeatures, objectIds, pbrMaterial} = this.props;
     const materialParser = this.parseMaterial(pbrMaterial, mesh);
     const shaders = this.getShaders();
     validateGeometryAttributes(mesh.attributes);
     const model = new Model(this.context.gl, {
       ...this.getShaders(),
-      id: this.props.id,
+      id,
       geometry: mesh,
       defines: {...shaders.defines, ...materialParser?.defines},
       parameters: materialParser?.parameters,
       isInstanced: true
     });
 
+    model.setUniforms({
```

Comment:
- This should be set in `updateStates` or `draw`
- Is `pickFeatures` necessary? Isn't `objectId` enough of an indicator to turn on this feature?

Reply:
1. Moved to the `draw` method.
2. Yes it is necessary because `objectIds` are created when we parse i3s node geometry in` loaders.gl` and if geometry has such object ids we put them to the tile content.  We need this flag to be able to switch between `instance` picking and `feature` picking. We can't decide to put `objectIds` or not on the loaders.gl side because we don't know what mode of picking will be selected by user in the `deck.gl` _MeshLayer.